orc8r/cloud/go/services/bootstrapper/servicers/registration/storage.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"github.com/pkg/errors"
+
+	"magma/orc8r/cloud/go/blobstore"
+	"magma/orc8r/cloud/go/services/bootstrapper"
+	"magma/orc8r/cloud/go/storage"
+	mErrors "magma/orc8r/lib/go/errors"
+	"magma/orc8r/lib/go/protos"
+)
+
+const networkWildcard = "*"
+
+type Store interface {
+	SetTokenInfo(oldNonce string, tokenInfo protos.TokenInfo) error
+	GetTokenInfoFromLogicalID(networkID string, logicalID string) (*protos.TokenInfo, error)
+	GetTokenInfoFromNonce(nonce string) (*protos.TokenInfo, error)
+	IsNonceUnique(nonce string) (bool, error)
+}
+
+type blobstoreStore struct {
+	factory blobstore.StoreFactory
+}
+
+func NewBlobstoreStore(factory blobstore.StoreFactory) Store {
+	return &blobstoreStore{factory}
+}
+
+func (b *blobstoreStore) SetTokenInfo(oldNonce string, tokenInfo protos.TokenInfo) error {
+	networkID := tokenInfo.GatewayPreregisterInfo.NetworkId
+	logicalID := tokenInfo.GatewayPreregisterInfo.LogicalId
+
+	store, err := b.factory.StartTransaction(nil)
+	if err != nil {
+		return err
+	}
+	defer store.Rollback()
+
+	logicalIDBlob, err := tokenInfoToBlob(bootstrapper.LogicalIDToTokenInfo, logicalID, tokenInfo)
+	if err != nil {
+		return err
+	}
+	err = store.Write(networkID, blobstore.Blobs{logicalIDBlob})
+	if err != nil {
+		return err
+	}
+
+	nonceBlob, err := tokenInfoToBlob(bootstrapper.NonceTokenToInfoMap, tokenInfo.Nonce, tokenInfo)
+	if err != nil {
+		return err
+	}
+	err = store.Write(networkWildcard, blobstore.Blobs{nonceBlob})
+	if err != nil {
+		return err
+	}
+
+	if oldNonce != "" {
+		oldNonceTK := storage.TKs{{
+			Type: string(bootstrapper.NonceTokenToInfoMap),
+			Key:  oldNonce,
+		}}
+		err = store.Delete(networkWildcard, oldNonceTK)
+		if err != nil {
+			if err != mErrors.ErrNotFound{
+				return err
+			}
+		}
+	}
```

Comment:
I wouldn't worry about deletion. Or, if you do wanna include it -- add it to a different method

Reply:
Why not delete?

Reply:
1. It's just a cache, we can handle cache cleanup separately if it becomes an issue, 2. It feels cleaner to separate them out -- it might feel surprising that a "get token" call would delete things

In either case, if you wanna include the "delete old token", let's just add a separate store method to handle it -- since it doesn't need to be in the same tx as the get/set